### PR TITLE
Updated and reviewed till end of ch 12

### PR DESCRIPTION
**Pull request by mishaone**
* Updated a lot of content
* Removed all images that were not essential to understanding the instructions
* Remove a lot of conditionals to make text simpler

### CODE REVIEW

File: docs/topics/modules/forking_example_repository.adoc [hunk 1]
```diff
@@ -1,24 +1,22 @@
...
-In addition to creating projects based on quickstarts, you can also add projects by using the *Import an existing codebase* feature.
-
+In addition to creating projects using quickstarts, you can add an existing proeject to {osio} by importing the codebase.
 
 [discrete]
 == Prerequisites
 
-Reset your {osio} environment by following instructions in <<resetting_your_osio_account>>.
-
+* Ensure that you have reset your {osio} environment by following the instructions in <<resetting_your_osio_account>>.
```
Comment (Preeticp) [inline docs/topics/modules/forking_example_repository.adoc]: that you reset

File: docs/topics/modules/reviewing_publishing_changes.adoc [hunk 2]
```diff
@@ -13,23 +13,21 @@ ifeval::["{context}" == "user-guide"]
...
 +
-** Different versions of your application are now deployed to *Stage* and *Run*. Version 1.0.2 of the application, which includes your committed change to the code, is deployed to *Stage* because you have not yet promoted it to *Run*. The older version, 1.0.1, is deployed to *Run* because you approved it the last time the pipeline build executed.
+** Different versions of your application are now deployed to *Stage* and *Run*. Version *1.0.2* of the application, which includes your committed change to the code, is deployed to *Stage* because you have not yet promoted it to *Run*. The older version, *1.0.1*, is deployed to *Run* because you approved it the last time the pipeline build executed.
 ** The green check marks indicate that both builds are operational.
-** The *1 pod* indicates that each of the application builds is scaled to one pod in OpenShift Online. The number of pods indicates the number of running instances of the application.
+** The *1 pod* indicates that each of the application builds scales to one pod in OpenShift Online. The number of pods indicates the number of running instances of the application.
```
Comment (Preeticp) [inline docs/topics/modules/reviewing_publishing_changes.adoc]: scale

File: docs/topics/modules/reviewing_publishing_changes.adoc [hunk 3]
```diff
@@ -38,7 +36,7 @@ image::expand_version.png[Expand the application version]
 . At the *Approve* stage, click btn:[Input Required].
 . Click btn:[Promote] to promote version 1.0.2 of the application to *Run*.
 +
-Your changes are now available on both *Stage* and *Run*. If you return to the *Deployments* tab, you can see the version for both *Stage* and *Run* are now 1.0.2.
+Your changes are now available on both *Stage* and *Run*. If you return to the *Deployments* tab, you can see the version for both *Stage* and *Run* are now version 1.0.2 of the application.
```
Comment (Preeticp) [inline docs/topics/modules/reviewing_publishing_changes.adoc]: Suggest: If you return to the *Deployments* tab, you can see version 1.0.2 of the application deployed to *Run*.

File: docs/topics/modules/running_your_project_in_the_che_workspace.adoc [hunk 4]
```diff
@@ -26,25 +25,16 @@ See <<create_run_command_macro>> for instructions on setting up a run command.
 +
 . A *run* terminal appears at the bottom pane of the Che workspace. When the `mvn{nbsp}build` command finishes executing, the *run* view displays the *Succeeded in Deploying verticle* message:
 +
-image::{context}_deployed_verticle.png[Successfully deployed verticle]
+----
+[INFO] INFO: Succeeded in deploying verticle
```
Comment (Preeticp) [inline docs/topics/modules/running_your_project_in_the_che_workspace.adoc]: Do we need this in addition? The step tells the user about the displayed message. 
Deploying>deploying.

**Comment by Preeticp** [comment]
@mishaone reviewed and added suggestions.

File: docs/topics/modules/creating_che_workspace.adoc [hunk 5]
```diff
@@ -22,25 +22,21 @@ endif::[]
 
 // for importing-existing-project
 ifeval::["{context}" == "importing-existing-project"]
-Create a new <<about_workspaces,Che workspace>> for your project:
+To create a new <<about_workspaces,Che workspace>> for your project to edit your code:
```
[the remark this answers is not in the document]
Comment (mishaone) [reply docs/topics/modules/creating_che_workspace.adoc]: Fixed.

File: docs/topics/modules/creating_che_workspace.adoc [hunk 6]
```diff
@@ -22,25 +22,21 @@ endif::[]
...
 ====
 If you see a message about the workspaces loading (see screen capture), it indicates that your Che instance is idle. Allow several minutes for the loading process to finish.
 
 image::loading_workspaces.png[Loading Workspaces]
 ====
 
-. When your workspace is ready, it automatically loads in a new tab. Alternatively, click btn:[Open] to view your new Che workspace in a new browser tab. This workspace loads the codebase for the listed project.
-+
-image::new_ws.png[New Workspace]
+. When your workspace is ready, it automatically loads in a new tab. If the workspace does not automatically load in a new browser tab, click btn:[Open] to view your new Che workspace in a new browser tab. This workspace loads the codebase for the listed project.
```
[the remark this answers is not in the document]
Comment (mishaone) [reply docs/topics/modules/creating_che_workspace.adoc]: Left the first two, removed the third and deleted the last part of the sentence.

File: docs/topics/getting-started-guide.adoc [hunk 7]
```diff
@@ -17,10 +17,10 @@ include::modules/planning_your_work.adoc[leveloffset=+1]
 
 include::optimizing_memory_usage.adoc[leveloffset=+1]
 
-include::analyzing-your-application.adoc[leveloffset=+1]
-
 include::using_advanced_che_features.adoc[leveloffset=+1]
 
+include::analyzing-your-application.adoc[leveloffset=+1]
```
[the remark this answers is not in the document]
Comment (mishaone) [reply docs/topics/getting-started-guide.adoc]: For the life of me, I can't remember what the reason was for moving it here. :/

Moving it back. Maybe it'll come back to me later.

File: docs/topics/modules/forking_example_repository.adoc [hunk 8]
```diff
@@ -1,24 +1,22 @@
 [id="forking_example_repository"]
 = Forking an example repository
 
-In addition to creating projects based on quickstarts, you can also add projects by using the *Import an existing codebase* feature.
-
+In addition to creating projects using quickstarts, you can add an existing proeject to {osio} by importing the codebase.
```
[the remark this answers is not in the document]
Comment (mishaone) [reply docs/topics/modules/forking_example_repository.adoc]: MInor fixes to address this feedback.

File: docs/topics/modules/forking_example_repository.adoc (hunk 1 above)
Comment (mishaone) [reply docs/topics/modules/forking_example_repository.adoc]: s/Ensure that you have reset/Reset 
This makes it more active.

File: docs/topics/modules/reviewing_publishing_changes.adoc (hunk 2 above)
Comment (mishaone) [reply docs/topics/modules/reviewing_publishing_changes.adoc]: Fixed

File: docs/topics/modules/reviewing_publishing_changes.adoc (hunk 3 above)
Comment (mishaone) [reply docs/topics/modules/reviewing_publishing_changes.adoc]: Changed slightly to accommodate feedback

File: docs/topics/modules/running_your_project_in_the_che_workspace.adoc (hunk 4 above)
Comment (mishaone) [reply docs/topics/modules/running_your_project_in_the_che_workspace.adoc]: Slightly changing text. I think the actual message appended to the end is useful in terms of verification so removing that specific text from the line before it.

**Comment by mishaone** [comment]
Thanks, preeti. Should be all done. Committing and merging changes now.

**Comment by mishaone** [comment]
@rkratky @Preeticp this seems to happen for me a lot. Can either of you point out why it's incorrect if my **git pull upstream master** and then **git rebase** incorporates Preeti's comments? I think Preeti called this out as something that should not be happening. My workflow is essentially this:
1. make changes, commit in new branch.
2. check out master, git pull upstream master
3. check out new branch, git rebase to resolve any merge conflicts
4. git push origin newbranch

I think seem to get the following error:
**error: failed to push some refs to 'https://github.com/mishaone/fabric8-online-docs'
hint: Updates were rejected because the tip of your current branch is behind
hint: its remote counterpart. Integrate the remote changes (e.g.
hint: 'git pull ...') before pushing again.
hint: See the 'Note about fast-forwards' in 'git push --help' for details.**

So then I do the following:
1. git branch --set-upstream-to=origin/issue-123 issue-123
2. git pull
3. git push origin issue-123

So now I have Preeti's commits included with mine but this is a bad thing?

**Comment by Preeticp** [comment]
I am not sure about it being bad to have my commits in yours, but afaik we should have only our commits in our PR. I am sure Robert would have exact and detailed solution, but here is my workflow, in case that helps:
Normal workflow:
1. git checkout master
2. git fetch upstream
3. git merge upstream/master
4. git checkout -b
5. Work
6. git add <doc file>.adoc or git add .
7. git commit. Add commit mssg
8. git push origin <local_branch_name>
9. git checkout master
10. git push origin master

**Comment by Preeticp** [comment]
Incase of merge conflicts: 
    1. cd <Repo you are working on>
    2. git checkout master
    3. git fetch upstream
    4. git merge upstream/master  or  
    5. git checkout <the branch you worked earlier on>
    6. git rebase master 
       – In case of merge conflicts- resolve the conflict and 
    7. git add <path to file>
    8. git rebase --continue
    9. git commit --amend
    10. git push origin <the branch you worked earlier on>. If rejected use `git push origin <local_branch_name> -f` to push forcefully.
    11. git checkout master
    12. git push origin master

**Comment by mishaone** [comment]
@Preeticp I didn't care about this before because I assumed the commits includes are playing your (merged) changes over mine so that we have the appropriate changes. Do you know is these included commits are already merged or not? If not, they shouldn't appear in my commits at all (since they are not yet merged and therefore not approved) and if they are, it shouldn't be a problem to have them explicitly included anyway. Thoughts? I don't really have enough git expertise to know if this is bad, but it also takes a while to redo the changes so I thought I should clarify and fix my workflow if there is something I'm habitually doing wrong.

**Comment by Preeticp** [comment]
Yes I merged these commits today morning. About the rest Robert would probably be the best person to answer. It's good you asked, cause it helps all of us to know answers to such questions.

**Comment by mishaone** [comment]
I'll wait for @rkratky to comment then. Thanks, Preeti.

**Comment by rkratky** [comment]
@mishaone Your workflow seems overly complicated and error-prone to me. I suggest trying:

**Get your fork (and your local clone of the fork) up-to-date with upstream:**

0. `git checkout master`
1. `git fetch upstream`
2. `git merge upstream/master` (I prefer this to `git pull` because it gives you a chance to react to potential conflicts -- though there should never ever be any conflicts as long as you never work directly on `master`.)
3. `git push`

**Work on your issue:**

1. `git checkout -b i123-f8-branch-name`
2. `git commit -m "Blah blah" file.adoc` (repeat as needed)
3. `git push --set-upstream origin i123-f8-branch-name`

**Submit a new PR and fix stuff based on reviews:**

1. Submit PR from `i123-f8-branch-name` through GH UI.
2. Get reviews.
3. `git checkout i123-f8-branch-name` (unless you're already there)
4. `git commit -m "Blah blah" file.adoc` (repeat as needed)
5. `git push` (GH automagically adds your latest commits not just to your branch on the fork but also to the open PR.)

............................
If someone else has been meddling with the PR (i.e. pushed commits to your working branch (`i123-f8-branch-name`), then you need to make sure you update you local copy of the repo/branch before committing your own fixes.

The above becomes:

3. `git checkout i123-f8-branch-name` (unless you're already there)
4. `git fetch`
5. `git merge origin/i123-f8-branch-name`
6. `git commit -m "Blah blah" file.adoc` (repeat as needed)
5. `git push`

And finally, if you've committed some fixes (but not pushed them) and the other person pushed their commits in the meantime, you may get conflicts during the merge in step 5 above. The best way is to work to resolve the conflicts.

You can also rebase (i.e. grab the other person's commits, apply them on the original state of the branch, and then 'replay' your local commits on top of that, thus preventing (most) conflicts). However, rebasing recreates history, so if you expect to continue working on the branch with other people, it screws up everyone else (when you push your rebased branch, and they fetch it to update their local copy, their old commits will suddenly have different hashes because you recreated them, thus creating a big conflict for them).
............................

**Merge completed work and cover your tracks:**

1. Merge PR.
2. Click on **Delete branch** button on the page with the merged PR. This deletes the branch from your fork.
3. `git branch -D i123-f8-branch-name` (deletes the branch from your local copy)

Rinse and repeat.

**Comment by Preeticp** [comment]
Thanks for the explanation @rkratky, the second last section was totally new to me and I will start implementing the last one about deleting branches.

**Comment by mishaone** [comment]
+1 thanks @rkratky. That clears up a lot of the questions I had around this issue. I inherited the old workflow from devstudio and kept using it assuming it was the simplest way to go (and resulted in few issues, if any). I'll redo the changes in a new branch and merge them. This includes all the changes in this thread so we should have the appropriate version merged. I'll also update my workflow to match these instructions from now on.

**Comment by mishaone** [comment]
Sorry folks, it's really frustrating but I've royally screwed this up now in trying to recover without just manually applying all the changes again to a new local copy of the repo. Scrapping this and manually applying changes tomorrow. This won't be completed in time for reporting this week as I'd hoped.

**Comment by Preeticp** [comment]
Closing this PR due to the many merge conflicts both from 3 of Misha's earlier PRs as well as my last PR. All the content and screenshot changes have been pushed and merged through fresh PRs: #422 and #423 respectively.

**Pull request closed without merging**